src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -184,6 +184,10 @@ public FleetStatusService(Topics topics, MqttClient mqttClient, DeploymentStatus
         this.mqttClient.addToCallbackEvents(callbacks);
 
         TestFeatureParameters.registerHandlerCallback(this.getName(), this::handleTestFeatureParametersHandlerChange);
+
+        if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
+            this.isConnected.set(false);
```

Comment:
How does this affect FSS? Make sure you understand the ramifications of it. For example, do we cache data in memory while waiting for a connection, could we blow up our memory usage because of this?

Reply:
So FSS data is computed ad-hoc and not cached. It just skips the computation if it's not connected.
Also it looks like we can skip the scheduled task and listeners setup here because right now connection can't be brought online without a restart.